### ISSUE TITLE
Controllers in added paths

### ISSUE THREAD

**Issue by agentphoenix** [fuel/core#76]
I'm not sure if this is a bug or the intended behavior, but when I add a path to Fuel for inclusion in the file system (through `Fuel::add_path()`), I can't use controllers in the added path location. I tossed a controller in to the added path's `classes/controller` directory and went there in my browser only to be met by a Fuel exception (**Fuel\Core\Fuel_Exception [ Error ]: It appears your _404_ route is incorrect. Multiple Recursion has happened.**).

At this point, my controller is dead simple as I'm just trying to get it to print something out:

``` php
<?php

class Controller_Welcome extends Controller {

    public function before()
    {
        parent::before();
    }

    public function action_index()
    {
        echo 'Foo!';
    }
}
```

Is this a bug or expected behavior? If it's the latter, I'm just curious why controllers aren't allowed in those added paths. Thanks!

**Comment by jschreuder** [comment]
find_file() can't know whether a file is namespaced or not, as such it's impossible for it to know whether a file from some path is valid class or not. This is why the controller loading uses class_exists() instead, which fires the autoloader. If you want controllers from a package you need to either have that package namespaced and its namespace as a 'core extension' or register the controller with the autoloader without namespace (and don't namespace the file either).
In other words: a controller needs to be available in the core namespace to be loaded by the autoloader.
